pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -1603,7 +1603,7 @@ func (j *jsonPopulateRecordSetGenerator) Values() (tree.Datums, error) {
 }
 
 type checkConsistencyGenerator struct {
-	db       *kv.DB
+	txn      *kv.Txn
```

Comment:
this change is suspicious, no? Are you even allowed to run these transactionally?

Reply:
The answer is that this thing should be abstracted through an interface.

Reply:
Yeah this is suspect.

What do you propose for the interface? I don't really have a great idea here, best I can think of is make some "DBRunner" interface which will have to go on EvalContext to be passed into this `makeCheckConsistencyGenerator`, not sure where this package would go in and make `kv.DB` implement it. 

Actually that doesn't work either since the `DBRunner` would need to import kv for `kv.Batch`

Reply:
Nvm, woke up with better ideas